helper/restore_helper.go

```diff
@@ -102,14 +107,51 @@ func (r *RestoreReader) positionReader(pos uint64, oid int) error {
 	return nil
 }
 
+func (r *RestoreReader) discardData(num int64) (int64, error) {
+	if r.readerType != SUBSET {
+		panic("discardData should be called for readerType == SUBSET only")
+	}
+
+	if r.discardErr {
+		logVerbose(fmt.Sprintf("%d bytes to discard, but discard error has already occurred. Don't read", num))
+		return 0, discardError
+	}
+
+	n, err := io.CopyN(io.Discard, r.bufReader, num)
+	if err == nil {
+		logVerbose(fmt.Sprintf("discarded %d bytes", n))
+	} else {
+		r.discardErr = true
+		err = fmt.Errorf("discarded %d bytes from %d: [%w]", n, num, err)
```

Comment:
Is it possible to avoid nesting in the error text? Again using Join...

Reply:
> what is the problem of current version?

Nested error texts look terrible

> discard error in copyData: [copied %d bytes from %d: [%w]]

and in tests you are forced to check these nestings.

Reply:
replaced with errors.Wrapf